File: lab_2/generator.cpp
#include <iostream>
#include <ctime>

using namespace std;

void generating_binary_sequence(int n) {
 /**
 * Function initializes the random number generator using the current time and generates
 * n-bit binary sequence, printing it to the standard output.
 * @param n - number of bits in sequence
 * @return None
 */
    srand(static_cast<unsigned>(time(0)));

    for (int i = 0; i < n; ++i) {
        int bit = rand() % 2;
        cout << bit;
    }
}


int main() {
 /**
 * The main function calls generating_binary_sequence to generate a random binary sequence
 * and print it to the standard output.
 * @return The program exit code.
 */
    int n;
    cout << "Enter number of bits for sequence: "; // we need use 128-bit sequence
    cin >> n;
    generate_binary_sequence(n);
    return 0;
}

File: lab_2/generator.java
import java.util.Scanner;

public class generator {
/**
* Class for generating a pseudo-random the specified length binary sequence
*/
    public static void main(String[] args) {
     /**
     * The main entry point of the program. Generates and prints the specified length binary sequence
     * @param args - command line arguments.
     */
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter number of bits for sequence: ");
        int length = scanner.nextInt(); // we need use 128-bit sequence
        StringBuilder binarySequence = new StringBuilder();
        for (int i = 0; i < length; i++) {
            int randomBit = (int) (Math.random() * 2);
            binarySequence.append(randomBit);
        }

        System.out.println(binarySequence.toString());
    }
}